### PR TITLE
Remove legacy API test that already have coverage and refactor code.

### PR DESCRIPTION
**Pull request by codemzs**
Deleted all tests under test/Microsoft.ML.Tests/Scenarios/PipelineApi and removed SentimentPredictionTests.cs from test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation because these tests already have equivalent tests that use new api under test/Microsoft.ML.Tests/Scenarios/Api/Estimators. 

Verified the code coverage wasn't affected too much, its now at **68.94% vs 68.97%** the difference is in the below dlls:
microsoft.ml.data.dll - 0.05%
microsoft.ml.entrypoints.dll - 0.18% (expected since legacy api calls into core entrypoints API)
microsoft.ml.fasttree.dll - 0.04%
microsoft.ml.hallearners.dll - 0.99% (This drop is coming from the removal of SymSGD test, will add one in another PR)
microsoft.ml.transforms.dll - 0.06%

### CODE REVIEW

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs [hunk 1]
```diff
@@ -17,6 +17,9 @@ namespace Microsoft.ML.Scenarios
 {
     public partial class ScenariosTests
     {
+        public const string SentimentDataPath = "wikipedia-detox-250-line-data.tsv";
+        public const string SentimentTestPath = "wikipedia-detox-250-line-test.tsv";
+
```
Comment (abgoswam) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: perhaps this should go to some relevant file ?

**Comment by abgoswam** [comment]
> #pragma warning disable 612

so we should be able to delete this now i guess

---
Refers to: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/IrisPlantClassificationTests.cs:11 in acfd3b0. [](commit_id = acfd3b04807db8d1dcc38f558c2e9c7dfeb642ab, deletion_comment = False)

**Comment by abgoswam** [comment]
>         private void CompareMatrics(MultiClassClassifierMetrics metrics)

can we fix this typo in this PR itself 

---
Refers to: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/IrisPlantClassificationTests.cs:90 in acfd3b0. [](commit_id = acfd3b04807db8d1dcc38f558c2e9c7dfeb642ab, deletion_comment = False)

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/IrisPlantClassificationTests.cs [hunk 2]
```diff
@@ -102,39 +99,6 @@ private void CompareMatrics(MultiClassClassifierMetrics metrics)
...
-                using (var rep = RepositoryReader.Open(memoryStream, ch))
-                {
-                    IDataLoader testPipe = ModelFileUtils.LoadLoader(env, rep, new MultiFileSource(testDataPath), true);
-                    RoleMappedData testRoles = new RoleMappedData(testPipe, label: "Label", feature: "Features");
-                    return ScoreUtils.GetScorer(pred, testRoles, env, testRoles.Schema);
-                }
-            }
-        }
     }
 #pragma warning restore 612
```
Comment (abgoswam) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/IrisPlantClassificationTests.cs]: >#pragma warning restore 61 [](start = 0, length = 26)

delete ?

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs (hunk 1 above)
Comment (sfilipi) [reply test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: you can use TestDatasets.Sentiment.trainFile, and  TestDatasets.Sentiment.testFile

---
In reply to: [245177801](https://github.com/dotnet/machinelearning/pull/2015#discussion_r245177801) [](ancestors = 245177801)

**Pull request merged**